python/cuml/cuml/_thirdparty/sklearn/utils/sparsefuncs.py

```diff
@@ -148,17 +148,18 @@ def mean_variance_axis(X, axis):
     else:
         _raise_typeerror(X)
 
-
-ufunc_dic = {
-    'min': np.min,
-    'max': np.max,
-    'nanmin': np.nanmin,
-    'nanmax': np.nanmax
-}
+# this is a function to avoid attribute access at import time in CPU-only environments
+def ufunc_dic():
+    return {
+        'min': np.min,
+        'max': np.max,
+        'nanmin': np.nanmin,
+        'nanmax': np.nanmax
+    }
```

Comment:
This non-lazy attribute access at the module level was triggering import errors because in this module, `np.` is actually an alias for `cupy`, which is imported with `gpu_only_import()`:

https://github.com/rapidsai/cuml/blob/80b98b12e93cde6213111507fb2f920ffe29f4e3/python/cuml/cuml/_thirdparty/sklearn/utils/sparsefuncs.py#L24